src/main/java/com/puppycrawl/tools/checkstyle/checks/design/FinalClassCheck.java

```diff
@@ -253,14 +253,39 @@ public void leaveToken(DetailAST ast) {
      * @return true if given class should be declared as final otherwise false
      */
     private static boolean shouldBeDeclaredAsFinal(ClassDesc desc) {
-        return desc.isWithPrivateCtor()
+        return (desc.isWithPrivateCtor() || isDefaultConstructor(desc.getTypeDeclarationAst()))
                 && !(desc.isDeclaredAsAbstract()
                     || desc.isSuperClassOfAnonymousInnerClass())
                 && !desc.isDeclaredAsFinal()
                 && !desc.isWithNonPrivateCtor()
                 && !desc.isWithNestedSubclass();
     }
 
+    /**
+     * Check if class is private.
+     *
+     * @param desc description of class
+     * @return true if class is private
+     */
+    private static boolean isClassPrivate(DetailAST desc) {
```

Comment:
Put method it below first usage.
Move it after isDefaultConstructor.

Reply:
Done